backend/ee/danswer/server/tenants/api.py

```diff
@@ -60,3 +74,53 @@ def create_tenant(
     finally:
         if token is not None:
             current_tenant_id.reset(token)
+
+
+@router.post("/product-gating")
+def gate_product(
+    product_gating_request: ProductGatingRequest, _: None = Depends(control_plane_dep)
+) -> None:
+    """
+    Gating the product means that the product is not available to the tenant.
+    They will be directed to the billing page.
+    We gate the product when
+    1) User has ended free trial without adding payment method
+    2) User's card has declined
+    """
+    if not MULTI_TENANT:
```

Comment:
Is there a cleaner/easier way to check this at the relevant endpoints/modules than this? I think we would need to add this to all the MT stuff and it's 1. very easy to miss some, 2. poor code reuse

If there's not a better way then it's whatever

Reply:
Also for this one, if we do single tenant cloud I guess it's not needed? I'm assuming with those we'd have an enterprise contract?

Reply:
Think a better way is to conditionally add router to the main lifecycle– updated to reflect